### PR TITLE
[CARBONDATA-3902] Fix CDC delete data issue on partition table

### PR DESCRIPTION
**Pull request by Indhumathi27**
### Why is this PR needed?
 When only delete operation is executed through cdc merge command on partition table, loadMetaEntry of segments which has deleted rows, is not updated with latest tableupdatestatus filename. On query, will give wrong results. 
 
 ### What changes were proposed in this PR?
Update loadMetaEntry of segments which has deleted rows, with latest tableupdatestatus filename, after delete operation on partition table, if loadDF.count  == 0

NOTE: If merge has data to be inserted after(U & D), CarbonInsertIntoDf will update the tableupdatestatus file name in load meta entry, during load.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1656/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3398/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3400/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1658/

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/CarbonMergeDataSetCommand.scala [hunk 1]
```diff
@@ -175,6 +178,21 @@ case class CarbonMergeDataSetCommand(
...
         throw new CarbonMergeDataSetException("writing of update status file failed")
       }
+      if (carbonTable.isHivePartitionTable) {
+        // If load count is 0 and if merge action contains delete operation, update
+        // tableUpdateStatus file name in loadMeta entry
+        if (count == 0 && hasDelAction && !tuple._1.isEmpty) {
+          val loadMetaDataDetails = SegmentStatusManager.readTableStatusFile(CarbonTablePath
+            .getTableStatusFilePath(carbonTable.getTablePath))
+          CarbonUpdateUtil.updateTableMetadataStatus(loadMetaDataDetails.map(l =>
+            new Segment(l.getMergedLoadName,
```
Comment (akashrn5) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/CarbonMergeDataSetCommand.scala:188]: rename variable `l` to `loadMetadataDetails`

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala [hunk 2]
```diff
@@ -725,6 +725,45 @@ class MergeTestCase extends QueryTest with BeforeAndAfterAll {
...
+          StructField("newValue", StringType),
+          StructField("deleted", BooleanType), StructField("time", IntegerType))))
+    ccd.createOrReplaceTempView("changes")
+
+    ccd = sql("SELECT key, latest.newValue as newValue, latest.deleted as deleted FROM ( SELECT key, max(struct(time, newValue, deleted)) as latest FROM changes GROUP BY key)")
+
+    target.as("A").merge(ccd.as("B"), "A.key=B.key").
+      whenMatched("B.deleted=true").delete().execute()
+
+    assert(getDeleteDeltaFileCount("target", "0") == 0)
```
Comment (akashrn5) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala:762]: if delete operation is successful, then it should contain delete delta file, please check and add proper assert for delete delta file count

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3449/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1707/

File: integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/CarbonMergeDataSetCommand.scala [hunk 3]
```diff
@@ -175,6 +178,21 @@ case class CarbonMergeDataSetCommand(
         LOGGER.error("writing of update status file failed")
         throw new CarbonMergeDataSetException("writing of update status file failed")
       }
+      if (carbonTable.isHivePartitionTable) {
+        // If load count is 0 and if merge action contains delete operation, update
+        // tableUpdateStatus file name in loadMeta entry
+        if (count == 0 && hasDelAction && !tuple._1.isEmpty) {
+          val loadMetaDataDetails = SegmentStatusManager.readTableStatusFile(CarbonTablePath
+            .getTableStatusFilePath(carbonTable.getTablePath))
+          CarbonUpdateUtil.updateTableMetadataStatus(loadMetaDataDetails.map(loadMetadataDetails =>
```
Comment (akashrn5) [inline integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/CarbonMergeDataSetCommand.scala:187]:
```suggestion
          CarbonUpdateUtil.updateTableMetadataStatus(loadMetaDataDetails.map(loadMetadataDetail =>
```

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala [hunk 4]
```diff
@@ -725,6 +725,47 @@ class MergeTestCase extends QueryTest with BeforeAndAfterAll {
     checkAnswer(sql("select * from target order by key"), Seq(Row("c", "200"), Row("d", "3"), Row("e", "100")))
   }
 
+  test("check the ccd delete with partition") {
```
Comment (akashrn5) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala:728]:
```suggestion
  test("check the cdc delete with partition") {
```

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala [hunk 5]
```diff
@@ -725,6 +725,47 @@ class MergeTestCase extends QueryTest with BeforeAndAfterAll {
...
+    ).asJava, StructType(Seq(StructField("key", StringType), StructField("value", StringType))))
+
+    initframe.write
+      .format("carbondata")
+      .option("tableName", "target")
+      .option("partitionColumns", "value")
+      .mode(SaveMode.Overwrite)
+      .save()
+    val target = sqlContext.read.format("carbondata").option("tableName", "target").load()
+    var ccd =
```
Comment (akashrn5) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala:746]:
```suggestion
    var cdc =
```

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala [hunk 6]
```diff
@@ -725,6 +725,47 @@ class MergeTestCase extends QueryTest with BeforeAndAfterAll {
...
+        Row("b", null, true, 2),
+        Row("c", null, true, 3),
+        Row("e", "100", false, 6)
+      ).asJava,
+        StructType(Seq(StructField("key", StringType),
+          StructField("newValue", StringType),
+          StructField("deleted", BooleanType), StructField("time", IntegerType))))
+    ccd.createOrReplaceTempView("changes")
+
+    ccd = sql("SELECT key, latest.newValue as newValue, latest.deleted as deleted FROM ( SELECT key, max(struct(time, newValue, deleted)) as latest FROM changes GROUP BY key)")
```
Comment (akashrn5) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala:759]: same as above

File: integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala [hunk 7]
```diff
@@ -869,10 +910,15 @@ class MergeTestCase extends QueryTest with BeforeAndAfterAll {
     CarbonProperties.getInstance().addProperty("carbon.enable.auto.load.merge", "false")
   }
 
-  private def getDeleteDeltaFileCount(tableName: String, segment: String): Int = {
+  private def getDeleteDeltaFileCount(tableName: String,
+      segment: String,
```
Comment (akashrn5) [inline integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/merge/MergeTestCase.scala:914]: no need to change the method signature, you are already having the carbonTable object, check from it whether its partition table or not.

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3458/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1716/

**Comment by akashrn5** [comment]
LGTM

**Pull request closed without merging**